packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -75,28 +75,44 @@ mixin RenderProxyBoxMixin<T extends RenderBox> on RenderBox, RenderObjectWithChi
   double computeMinIntrinsicWidth(double height) {
     if (child != null)
       return child!.getMinIntrinsicWidth(height);
-    return 0.0;
+    final BoxConstraints constraints = height.isInfinite
+      ? const BoxConstraints()
+      : BoxConstraints.tightFor(height: height);
```

Comment:
I always thought of the height passed to computeMinIntrinsicWidth to be more of a maxHeight - not a tight height...

If that's true, it make me wonder if we can actually use computeSizeForNoChild here since we do not know the minHeight.

@Hixie do you have thoughts on this?

Reply:
It's not a constraint at all. If the layout algorithm is strictly height-in-width-out, or is height-in-width-out when the width is unconstrained, then the height argument is the height to use. If the layout algorithm is independent of the context (e.g. it always tries to be a particular size), or if the layout algorithm is width-in-height-out, or if the layout algorithm uses both the incoming width and height constraints (e.g. it always sizes itself to BoxConstraints.biggest), then the height argument should be ignored.

Reply:
Right, but since this is `ProxyBox`, it doesn't know the layout algorithm - it just passes it to its child... except when it doesn't have a child, it makes sense that the intrinsic size pulls from `computeSizeForNoChild()`, which takes constraints.  So the operative question is: what to do with the height argument that's passed in:

1. use it to construct tight-height constraints for `computeSizeForNoChild()`
1. use it construct max-height constraints for `computeSizeForNoChild()`
1. pass unbounded constraints to `computeSizeForNoChild()`

Reply:
> it makes sense that the intrinsic size pulls from `computeSizeForNoChild()`

Why?

Reply:
'cause that's basically how `computeSizeForNoChild ()` is defined ("Calculate the size the [RenderProxyBox] would have under the given [BoxConstraints] for the case where it does not have a child.")...

Not doing so means you have cases where `computeSizeForNoChild ()` and the intrinsic sizes don't match at all.  This is what we have now, where e.g. `RenderCustomPaint` returns its preferred size in `computeSizeForNoChild ()` but Size.zero in intrinsic size.

Reply:
Ok, I re-did this PR to just focus on the problem with `RenderCustomPaint` -- that it wasn't using its preferred size in its intrinsic calculations.

PTAL.